tempesta_fw/cache.c

```diff
@@ -248,17 +261,120 @@ tfw_cache_employ_req(TfwHttpReq *req)
 	/* cache_fulfill - work as usual in cache mode. */
 	BUG_ON(cmd != TFW_D_CACHE_FULFILL);
 
+	if (req->cache_ctl.flags & TFW_HTTP_CC_NO_CACHE)
+		return false;
+
+	/* Assume there's no Date: header in the request. */
+	req->cache_ctl.timestamp = tfw_current_timestamp();
 	return true;
 }
 
+static inline bool
+tfw_cache_status_bydef(TfwHttpResp *resp)
+{
+	/*
+	 * TODO: Add 206 (Partial Content) status. Requires support
+	 * of incomplete responses, Range: and Content-Range: header
+	 * fields, and RANGE request method.
+	 */
+	switch (resp->status) {
+	case 200: case 203: case 204:
+	case 300: case 301:
+	case 404: case 405: case 410: case 414:
+	case 501:
+		return true;
+	}
+	return false;
+}
+
 static bool
-tfw_cache_employ_resp(TfwHttpReq *req)
+tfw_cache_employ_resp(TfwHttpReq *req, TfwHttpResp *resp)
 {
+	static const unsigned int __read_mostly cacheit =
+		TFW_HTTP_CC_HDR_EXPIRES | TFW_HTTP_CC_MAX_AGE
+		| TFW_HTTP_CC_S_MAXAGE | TFW_HTTP_CC_PUBLIC;
+	static const unsigned int __read_mostly authcan =
+		TFW_HTTP_CC_S_MAXAGE | TFW_HTTP_CC_PUBLIC
+		| TFW_HTTP_CC_MUST_REVAL | TFW_HTTP_CC_PROXY_REVAL;
```

Comment:
Usually defines are used in kernel for such masks - technically should be the same, but just more convenient form.